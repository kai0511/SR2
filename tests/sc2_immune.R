require(SC2)

lambda1 <- 15.5 # original value 960.5335 
lambda2 <- 0.6 # original value 10, 0.8, 0.5  

# ratio <- 0.05 # ratio for randomly sampling a proportion of observations for model selection
ratio <- 1 

cfd <- read.csv("/home/kai/data/single_cell_datasets/PMC7612735/global_cfd_meta.csv", row.names = 1)
load("~/data/single_cell_datasets/PMC7612735/global_2000hvg.RData")

if(ratio <  1){
    data_list <- lapply(data_list, function(d){
        obs_num <- ncol(d)
        idx <- sample(obs_num, floor(obs_num * ratio))
        d[, idx]
    })
}

dataset <- as.matrix(t(do.call(cbind, data_list)))
confounders <- as.matrix(cfd[rownames(dataset), 4:3])


sc_factor_num <- 23
# sc_factor_num <- seq(15, 40, by = 2)

# object <- SC2(dataset, confounders, split_ratio = 0.1, tuning_iter = 20)
# tune(object, as.integer(sc_factor_num), lambda1 = seq(2, 20, length.out = 5), lambda2 = seq(0.4, 0.8, by = 0.1))
object <- SC2(dataset, confounders, split_ratio = 0.1, tuning_iter = 20, global_tol = 1e-8, max_iter = 500)
rm(dataset, confounders); gc();
object <- fit(object, as.integer(sc_factor_num), batch_num = 60, is_batch = T, lambda1 = lambda1, lambda2 = lambda2)
save(object, file = paste0("SC2_immune_R", sc_factor_num, "_batch_fitted_object.RData"))
