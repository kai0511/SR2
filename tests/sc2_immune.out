Loading required package: SC2
Loading required package: RcppArmadillo
Loading required package: igraph

Attaching package: ‘igraph’

The following objects are masked from ‘package:stats’:

    decompose, spectrum

The following object is masked from ‘package:base’:

    union

Loading required package: RANN
             used   (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells     781255   41.8    1584765    84.7     782925    41.9
Vcells 1294627004 9877.3 2328496485 17765.1 1941506535 14812.6
Iteration 0 ---------------------------------
Train RMSE for iter 0:0.417615323385
Loss for iter 0:1003116.32049
Delta loss for iter 0:1.79769313486e+308
Iteration 10 ---------------------------------
Train RMSE for iter 10:0.414703537708
Loss for iter 10:985213.008667
Delta loss for iter 10:17903.3118261
Iteration 20 ---------------------------------
Train RMSE for iter 20:0.414664354701
Loss for iter 20:983562.756172
Delta loss for iter 20:1650.25249461
Iteration 30 ---------------------------------
Train RMSE for iter 30:0.414663568237
Loss for iter 30:982900.121535
Delta loss for iter 30:662.634637124
Iteration 40 ---------------------------------
Train RMSE for iter 40:0.414677248393
Loss for iter 40:982475.395316
Delta loss for iter 40:424.726218763
Iteration 50 ---------------------------------
Train RMSE for iter 50:0.414692515158
Loss for iter 50:982259.840043
Delta loss for iter 50:215.555273073
Iteration 60 ---------------------------------
Train RMSE for iter 60:0.414699114534
Loss for iter 60:982158.465252
Delta loss for iter 60:101.374791356
Iteration 70 ---------------------------------
Train RMSE for iter 70:0.414701067209
Loss for iter 70:982092.637195
Delta loss for iter 70:65.8280565221
Iteration 80 ---------------------------------
Train RMSE for iter 80:0.414701855343
Loss for iter 80:982041.845837
Delta loss for iter 80:50.7913580999
Iteration 90 ---------------------------------
Train RMSE for iter 90:0.414702163201
Loss for iter 90:981998.815116
Delta loss for iter 90:43.0307209237
Iteration 100 ---------------------------------
Train RMSE for iter 100:0.414702292852
Loss for iter 100:981962.602235
Delta loss for iter 100:36.2128807779
Iteration 110 ---------------------------------
Train RMSE for iter 110:0.414702035143
Loss for iter 110:981932.313844
Delta loss for iter 110:30.2883913342
Iteration 120 ---------------------------------
Train RMSE for iter 120:0.414701095539
Loss for iter 120:981906.299424
Delta loss for iter 120:26.014420496
Iteration 130 ---------------------------------
Train RMSE for iter 130:0.414699687311
Loss for iter 130:981883.950131
Delta loss for iter 130:22.3492922749
Iteration 140 ---------------------------------
Train RMSE for iter 140:0.414697844175
Loss for iter 140:981864.571414
Delta loss for iter 140:19.3787168004
Iteration 150 ---------------------------------
Train RMSE for iter 150:0.414695932169
Loss for iter 150:981846.655836
Delta loss for iter 150:17.9155786365
Iteration 160 ---------------------------------
Train RMSE for iter 160:0.414694266883
Loss for iter 160:981831.27286
Delta loss for iter 160:15.3829763033
Iteration 170 ---------------------------------
Train RMSE for iter 170:0.414692846609
Loss for iter 170:981816.397462
Delta loss for iter 170:14.8753975538
Iteration 180 ---------------------------------
Train RMSE for iter 180:0.414691571105
Loss for iter 180:981802.619415
Delta loss for iter 180:13.7780470268
Iteration 190 ---------------------------------
Train RMSE for iter 190:0.414690584701
Loss for iter 190:981790.354458
Delta loss for iter 190:12.2649564616
Iteration 200 ---------------------------------
Train RMSE for iter 200:0.414689741789
Loss for iter 200:981778.438208
Delta loss for iter 200:11.9162506053
Iteration 210 ---------------------------------
Train RMSE for iter 210:0.414689098203
Loss for iter 210:981766.667931
Delta loss for iter 210:11.7702765249
Iteration 220 ---------------------------------
Train RMSE for iter 220:0.414688846504
Loss for iter 220:981755.282443
Delta loss for iter 220:11.3854881609
Iteration 230 ---------------------------------
Train RMSE for iter 230:0.414688905982
Loss for iter 230:981744.292229
Delta loss for iter 230:10.9902142017
Iteration 240 ---------------------------------
Train RMSE for iter 240:0.414689081214
Loss for iter 240:981733.889669
Delta loss for iter 240:10.4025603489
Iteration 250 ---------------------------------
Train RMSE for iter 250:0.414689359612
Loss for iter 250:981724.854612
Delta loss for iter 250:9.03505660745
Iteration 260 ---------------------------------
Train RMSE for iter 260:0.41468961714
Loss for iter 260:981716.781043
Delta loss for iter 260:8.07356882328
Iteration 270 ---------------------------------
Train RMSE for iter 270:0.414689870713
Loss for iter 270:981710.44398
Delta loss for iter 270:6.33706366934
Iteration 280 ---------------------------------
Train RMSE for iter 280:0.414690084207
Loss for iter 280:981705.215628
Delta loss for iter 280:5.22835130314
Iteration 290 ---------------------------------
Train RMSE for iter 290:0.414690166222
Loss for iter 290:981700.846044
Delta loss for iter 290:4.36958389904
Iteration 300 ---------------------------------
Train RMSE for iter 300:0.414690244814
Loss for iter 300:981697.235216
Delta loss for iter 300:3.61082882062
Iteration 310 ---------------------------------
Train RMSE for iter 310:0.414690333365
Loss for iter 310:981693.833279
Delta loss for iter 310:3.40193628147
Iteration 320 ---------------------------------
Train RMSE for iter 320:0.414690331762
Loss for iter 320:981690.695931
Delta loss for iter 320:3.13734864409
Iteration 330 ---------------------------------
Train RMSE for iter 330:0.414690457967
Loss for iter 330:981688.089979
Delta loss for iter 330:2.60595201503
Iteration 340 ---------------------------------
Train RMSE for iter 340:0.414690493646
Loss for iter 340:981685.873242
Delta loss for iter 340:2.21673678723
Iteration 350 ---------------------------------
Train RMSE for iter 350:0.414690581812
Loss for iter 350:981683.493057
Delta loss for iter 350:2.38018489897
Iteration 360 ---------------------------------
Train RMSE for iter 360:0.414690736033
Loss for iter 360:981681.978761
Delta loss for iter 360:1.51429632993
Iteration 370 ---------------------------------
Train RMSE for iter 370:0.41469079485
Loss for iter 370:981680.488462
Delta loss for iter 370:1.49029817933
Iteration 380 ---------------------------------
Train RMSE for iter 380:0.414690907876
Loss for iter 380:981678.771107
Delta loss for iter 380:1.71735548496
Iteration 390 ---------------------------------
Train RMSE for iter 390:0.414690946738
Loss for iter 390:981677.411867
Delta loss for iter 390:1.35923963226
Iteration 400 ---------------------------------
Train RMSE for iter 400:0.414690976606
Loss for iter 400:981676.143329
Delta loss for iter 400:1.26853792951
Iteration 410 ---------------------------------
Train RMSE for iter 410:0.414691021319
Loss for iter 410:981674.823709
Delta loss for iter 410:1.3196205335
Iteration 420 ---------------------------------
Train RMSE for iter 420:0.414691019264
Loss for iter 420:981673.32297
Delta loss for iter 420:1.50073888362
Iteration 430 ---------------------------------
Train RMSE for iter 430:0.414691057963
Loss for iter 430:981672.790516
Delta loss for iter 430:0.532453720109
Iteration 440 ---------------------------------
Train RMSE for iter 440:0.414691054842
Loss for iter 440:981671.354745
Delta loss for iter 440:1.43577155122
Iteration 450 ---------------------------------
Train RMSE for iter 450:0.414691022519
Loss for iter 450:981670.306898
Delta loss for iter 450:1.0478466159
Iteration 460 ---------------------------------
Train RMSE for iter 460:0.414691078015
Loss for iter 460:981669.596093
Delta loss for iter 460:0.710805338575
Iteration 470 ---------------------------------
Train RMSE for iter 470:0.414691153106
Loss for iter 470:981668.987188
Delta loss for iter 470:0.608904348337
Iteration 480 ---------------------------------
Train RMSE for iter 480:0.414691160235
Loss for iter 480:981668.096536
Delta loss for iter 480:0.890652260045
Iteration 490 ---------------------------------
Train RMSE for iter 490:0.414691120339
Loss for iter 490:981667.018235
Delta loss for iter 490:1.07830076048
Iteration 500 ---------------------------------
Train RMSE for iter 500:0.414691141892
Loss for iter 500:981666.389722
Delta loss for iter 500:0.62851300207
